xray/commands/audit/sca/java/mvn_test.go

```diff
@@ -98,3 +100,135 @@ func TestMavenWrapperTrees(t *testing.T) {
 		}
 	}
 }
+
+func TestGetMavenPluginInstallationArgs(t *testing.T) {
+	args := GetMavenPluginInstallationGoals("testPlugin")
+	assert.Equal(t, "org.apache.maven.plugins:maven-install-plugin:3.1.1:install-file", args[0])
+	assert.Equal(t, "-Dfile=testPlugin", args[1])
+}
+
+func TestCreateSettingsXmlWithConfiguredArtifactory(t *testing.T) {
+	// Test case for successful creation of settings.xml.
+	mdt := MavenDepTreeManager{
+		DepTreeManager: &DepTreeManager{
+			server: &config.ServerDetails{
+				ArtifactoryUrl: "https://myartifactory.com/artifactory",
+				User:           "testUser",
+				Password:       "testPass",
+			},
+			depsRepo: "testRepo",
+		},
+	}
+	// Create a temporary directory for testing and settings.xml creation
+	tempDir := t.TempDir()
+	err := mdt.createSettingsXmlWithConfiguredArtifactory(tempDir)
+	assert.NoError(t, err)
+
+	// Verify settings.xml file creation with username and password
+	settingsXmlPath := filepath.Join(tempDir, "settings.xml")
+	actualContent, err := os.ReadFile(settingsXmlPath)
+	actualContent = []byte(strings.ReplaceAll(string(actualContent), "\r\n", "\n"))
+	assert.NoError(t, err)
+	expectedContent := `<?xml version="1.0" encoding="UTF-8"?>
+<settings xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.2.0 http://maven.apache.org/xsd/settings-1.2.0.xsd"
+          xmlns="http://maven.apache.org/SETTINGS/1.2.0"
+          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <server>
+        <id>artifactory</id>
+        <configuration>
+            <httpHeaders>
+                <property>
+                    <name>Authorization</name>
+                    <value>Basic dGVzdFVzZXI6dGVzdFBhc3M=</value>
+                </property>
+            </httpHeaders>
+        </configuration>
+    </server>
+    <mirrors>
+        <mirror>
+            <id>artifactory</id>
+            <url>https://myartifactory.com/artifactory/testRepo</url>
+            <mirrorOf>*</mirrorOf>
+        </mirror>
+    </mirrors>
+</settings>`
```

Comment:
For better readability, I suggest exporting the expectedContent to a class const